libs/auth-api-lib/src/lib/delegations/delegations.service.ts

```diff
@@ -538,27 +628,66 @@ export class DelegationsService {
         provider: DelegationProvider.PersonalRepresentativeRegistry,
       })
 
-      const rp = await this.prService.getByPersonalRepresentative(
+      const personalRepresentatives = await this.prService.getByPersonalRepresentative(
         user.nationalId,
         false,
       )
 
-      const resultPromises = rp.map(async (representative) =>
-        this.nationalRegistryClient
-          .getIndividual(representative.nationalIdRepresentedPerson)
-          .then((person) =>
-            toDelegationDTO(person?.name ?? 'Óþekkt nafn', representative),
+      const personPromises = personalRepresentatives.map(
+        ({ nationalIdRepresentedPerson }) =>
+          this.nationalRegistryClient.getIndividual(
+            nationalIdRepresentedPerson,
```

Comment:
One more (big) thing... The IDS needs to be as resilient as possible from outside dependencies like Þjóðskrá. If any of these `getIndividual` calls throw an Error (eg network issue, 500, some xroad error), we need to continue without crashing the request and try to keep as much functionality as possible.

Ideally we assume the person is still around and continue serving the delegation until proven otherwise.

It may complicate these functions some. Not sure what's the smartest way to tell apart "null" and the Error. Maybe just `.catch(error => return error)`, and work with an array of `Individual | null | Error`.

Sorry about mentioning this soo late ... I remember spotting this way earlier but somehow forgot to add it to my earlier reviews. 😭

Reply:
No problemo. I have update the code. Let's hope it does the trick.
Now we swallow the error and return appropriate items although the api call to Þjóðskrá errors. The fromName field is then populated as `"Óþekkt nafn"`

Reply:
These three commits are fixed to the newest comments.

https://github.com/island-is/island.is/pull/8402/commits/c7e6a20bb06a9a9bc26941f27c2d6aae0e7bcee1

https://github.com/island-is/island.is/pull/8402/commits/4f5276f35aab9cf9d9096a2a2d8ae4112c1325a5

https://github.com/island-is/island.is/pull/8402/commits/c7ead4539c7b95078d58f6461780cdc122022774

and the last commit is just a small refactor.

https://github.com/island-is/island.is/pull/8402/commits/33f46498ae7c3b933629d49132331a3956a45677